### ISSUE TITLE
`redundantReturn` rule not removing redundant parens in return statement

### ISSUE THREAD

**Issue by calda** [nicklockwood/SwiftFormat#1415]
We've found some cases where the `redundantReturn` rule isn't removing redundant parenthesis, for example in a return statement. Here are some test cases that currently fail on develop:

```swift
func testRedundantParensInReturnRemoved() {
    let input = "return (true)"
    let output = "return true"
    testFormatting(for: input, output, rule: FormatRules.redundantParens)
}

func testRedundantParensAroundTryExpressionRemoved() {
  let input = "return (try bar())"
  let output = "return try bar()"
  testFormatting(for: input, output, rule: FormatRules.redundantParens)
}
```

Is this a bug / oversight, or is this intentional?

**Comment by jparise** [comment]
Adding `"return"` to the rule's set of "conditionals" addresses the first case (`return (true)` → `return true`), but it seems like the second case (`return try (...)`) needs some additional support.

**Comment by nicklockwood** [comment]
I don't think it was intentional, but there may be some edges cases with return that I've forgotten about (e.g. if statement spans multiple lines and removing parens changes meaning).

With try, removing parens should also be fine, but have to be careful in try? case due to optional elision.
